vscode-car-plugin/src/main/java/org/wso2/maven/CARMojo.java

```diff
@@ -0,0 +1,230 @@
+
+    /**
+     * finalName to use for the generated capp project if the user wants to override the default name
+     *
+     * @parameter
+     */
+    private String finalName;
+
+    /**
+     * @parameter default-value="${project}"
+     */
+    private MavenProject project;
+
+    /**
+     * A classifier for the build final name
+     *
+     * @parameter
+     */
+    private String classifier;
+
+    public void execute() throws MojoExecutionException {
+        appendLogs();
+        // Create CApp
+        try {
+            // Create directory to be compressed.
+            String archiveDirectory = getArchiveFile("").getAbsolutePath();
+            boolean createdArchiveDirectory = org.wso2.developerstudio.eclipse.utils.file.FileUtils.createDirectory(
+                    archiveDirectory);
+
+            if (createdArchiveDirectory) {
+                CAppHandler cAppHandler = new CAppHandler();
+                List<ArtifactDependency> dependencies = new ArrayList<>();
+
+                cAppHandler.processConfigArtifactXmlFile(projectBaseDir, archiveDirectory, dependencies);
+                cAppHandler.processRegistryResourceArtifactXmlFile(projectBaseDir, archiveDirectory, dependencies);
+                cAppHandler.createDependencyArtifactsXmlFile(archiveDirectory, dependencies, project);
+
+                File fileToZip = new File(archiveDirectory);
+                zipFolder(fileToZip.getPath(), getArchiveFile(".car").getPath());
+
+                recursiveDelete(fileToZip);
+
+            } else {
+                getLog().error("Could not create corresponding archive directory.");
+            }
+        } catch (Exception e) {
+            getLog().error(e);
+        }
+    }
+
+    /**
+     * Append log of BUILD process when building started.
+     */
+    private void appendLogs() {
+        getLog().info("------------------------------------------------------------------------");
+        getLog().info("Building Synapse Config Project");
+        getLog().info("------------------------------------------------------------------------");
+    }
+
+    private File getArchiveFile(String fileExtension) {
+        File archiveFile;
+        if (finalName != null && !finalName.trim().equals("")) {
+            archiveFile = new File(archiveLocation, finalName + fileExtension);
+            return archiveFile;
+        }
+        String archiveFilename = project.getArtifactId() + "_" + project.getVersion() +
+                (classifier != null ? "-" + classifier : "") + fileExtension;
+        archiveFile = new File(archiveLocation, archiveFilename);
+
+        return archiveFile;
+    }
+
+    private void zipFolder(String srcFolder, String destZipFile) {
+        try {
+            FileOutputStream fileWriter = new FileOutputStream(destZipFile);
+            ZipOutputStream zip = new ZipOutputStream(fileWriter);
+
+            addFolderContentsToZip(srcFolder, zip);
+
+            zip.flush();
+            zip.close();
```

Comment:
Better to move this to finally block

Reply:
Corrected according to [try-with-resources Statement](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) in java